rfc/0054-sdk3-management-apis.md

```diff
@@ -2523,6 +2754,158 @@ interface AnalyticsIndex{
 }
 ```
 
+## AnalyticsLinkInterface
+
+`AnalyticsLink` provides a means of mapping analytics link details into an object.
+The interface methods are primarily designed for creation and modification of links.
+Note: Analytics link management is only supported for server 7.0+.
+
+```
+interface AnalyticsLink {
+  List<Byte> FormEncode();
+  
+  Validate();
+  
+  AnalyticsLinkType LinkType();
+}
+```
+
+### CouchbaseRemoteAnalyticsLink
+
+`CouchbaseRemoteAnalyticsLink` provides a means of mapping remote couchbase analytics link details into an object.
+The following fields are named according to how they are sent when form/json encoded.
+The object names should be implemented in a way idiomatic to the implementing language.
+
+* `dataverse` (string) - The dataverse that the link belongs to. Form can be one part `dataversename` or two parts `bucket.name/scope.name`.
+* `name` (string) - The name of the link.
+* `hostname` (string) - The hostname of the target couchbase cluster. This is `activeHostname` in the returned JSON payload for a get.
+* `encryption` (CouchbaseAnalyticsEncryptionSettings) - The encryption settings for the link. The contents of this field are flattened to the top level of the payload.
+* `username` (string) - The username to use for authentication.
+* `password` (string) - The password to use for authentication.
+
+A value for `type` of `couchbase` must be sent in the form payload, this is returned in the same manner in the JSON payload.
+This field is not exposed on the object and is either inferred from the object type or by calling `LinkType()`.
+
+The `Validate()` implementation must verify that:
+* `dataverse` is set.
+* `name` is set.
+* `hostname` is set.
+* When encryption level is set to "none" or "half":
+  * `username` is set.
+  * `password` is set.
+* When encryption level is set to "full":
+  * `certificate` is set.
+  * Either both `username` and `password` are set OR both `clientCertificate` and `clientKey` are set.
+
+The `LinkType()` implementation must return a value that corresponds to `"couchbase"`.
+
+On fetching a `CouchbaseRemoteAnalyticsLink`:
+  * The `hostname` property is present as `activeHostname`.
+  * The `password` property must be blanked out/left unset.
+  * The `clientKey` property must be blanked out/left unset.
+  * The `dataverse` property must be set by first checking the `dataverse` field in the response body and if empty falling back to the `scope` field.
```

Comment:
Is it possible for both the `dataverse` and `scope` fields to be present? To put it another way, does it matter which field we look for first?

Reply:
I don't believe that it is possible for both to be present so no I don't think that it does matter.